src/board.rs

```diff
@@ -486,6 +486,138 @@ impl Chessboard {
+                'b' => is_legal = Chessboard::legal_bishop(old_square, new_square),
+                'k' => is_legal = Chessboard::legal_king(old_square, new_square),
+                'q' => is_legal = Chessboard::legal_queen(old_square, new_square),
+                'n' => is_legal = Chessboard::legal_knight(old_square, new_square),
+                'P' => is_legal = Chessboard::legal_pawn(cb, old_square, new_square),
+                'R' => is_legal = Chessboard::legal_rook(old_square, new_square),
+                'B' => is_legal = Chessboard::legal_bishop(old_square, new_square),
+                'K' => is_legal = Chessboard::legal_king(old_square, new_square),
+                'Q' => is_legal = Chessboard::legal_queen(old_square, new_square),
+                'N' => is_legal = Chessboard::legal_knight(old_square, new_square),
+                _ => return false,
+            }
+        }
+        is_legal
+    }
+
+    fn legal_pawn(cb: &Chessboard, old_square: u64, new_square: u64) -> bool {
+        //if there is a piece diagonal
+
+        //if en passant
+
+        // if white turn and first position
+        if cb.white_turn && old_square < 15 && old_square > 7 {
+            new_square == old_square + 8 || new_square == old_square + 16
+        // if black turn and first position
+        } else if !cb.white_turn && old_square < 56 && old_square > 47 {
+            new_square == old_square - 8 || new_square == old_square - 16
+        //if white turn and not first position
+        } else if cb.white_turn {
+            new_square == old_square + 8
+        //if black turn and not first position
+        } else {
+            new_square == old_square - 8
+        }
+
+        //move 1 or 2 if on start square (and nothing in front of it)
+        //move 1 otherwise (and nothing in front of it)
+        //if opposing piece is diagonal to it
+        //if en passant is diagonal to it
+        //promote if at end
+    }
+    fn legal_knight(old_square: u64, new_square: u64) -> bool {
+        new_square == (old_square + 17)
+            || new_square == (old_square + 15)
+            || new_square == (old_square + 10)
+            || new_square == (old_square - 10)
+            || new_square == (old_square - 6)
+            || new_square == (old_square + 6)
+            || new_square == (old_square - 17)
+            || new_square == (old_square - 15)
+    }
+    fn legal_bishop(old_square: u64, new_square: u64) -> bool {
+        //add and subtract multiples of 7 and 9 (max 8 long diagonal)
+        if new_square > old_square {
+            (new_square - old_square) % 7 == 0 || (new_square - old_square) % 9 == 0
+        } else {
+            (old_square - new_square) % 7 == 0 || (old_square - new_square) % 9 == 0
+        }
+    }
+    fn legal_rook(old_square: u64, new_square: u64) -> bool {
+        if new_square % 8 == old_square % 8 || new_square / 8 == old_square / 8 {
+            return true;
+        }
+        false
+    }
```

Comment:
also smart. however we can remove the conditional and just return
`new_square % 8 == old_square % 8 || new_square / 8 == old_square / 8`